src/OrchardCore/Orchard.Environment.Shell.Abstractions/ShellSettings.cs

```diff
@@ -11,7 +11,6 @@ namespace Orchard.Environment.Shell
     /// </summary>
     public class ShellSettings
     {
-        private TenantState _tenantState = TenantState.Invalid;
```

Comment:
There is a setter, and if it's always invalid, why parse it again? You are wrong, stop fighting.